### PR TITLE
feat: implemented getting sfu-config data from resource manager

### PR DESCRIPTION
**Pull request by zapfire88**
Includes: 

- It is now possible to get sfu config data from a resource manager.
- Updated README with how to enable the use of the resource manager.

### CODE REVIEW

**Review by birme — CHANGES_REQUESTED** [review]
Looks good overall, but when thinking of this I would like to move this code to WhipEndpoint instance functions:

```
const edgeList = await whipEndpoint.readEdgeListFromFile(jsonFilePath);
```
and
```
const edgeList = await whipEndpoint.readEdgeListFromService(urlResourcemanager, territoryCode);
```
and then we can have
```
edgeList.forEach((pair) => whipEndpoint.registerBroadcasterClient({ client: new BroadcasterClient(pair.egressApiUrl), sfuUrl: pair.sfuApiUrl });
```

File: packages/server/src/whip/whipEndpoint.ts [hunk 1]
```diff
@@ -1,150 +1,181 @@
...
+    return <OriginsAndEdgesPerTerritory>JSON.parse(sfuConfigFileContents.toString());
+  }
+
+  async readEdgeListFromService(): Promise<OriginsAndEdgesPerTerritory> {
+    return (await callResourceManager(process.env.RESOURCE_MANAGER_URL)) as OriginsAndEdgesPerTerritory;
+  }
+
+  useResourceManager(): boolean {
+    if(this.resourceManagerUrl) {
+      console.log(`Env RESOURCE_MANAGER_URL is set. Getting sfu-config from Resource Manager.`);
```
Comment (birme) [inline packages/server/src/whip/whipEndpoint.ts:167]: This is not exactly true as you would set the resourceManagerUrl via constructor options and not the environment variable. Remember that the WHIP Endpoint is a library and the `server.ts` is just an example / reference implementation of how to use it. Anything that is specific to the server.ts implementation should not be in the library.

File: packages/server/src/whip/whipEndpoint.ts [hunk 2]
```diff
@@ -1,150 +1,181 @@
...
+  }
+
+  getServerAddress(): string | undefined {
+    if (this.hostname) {
+      return (this.useHttps ? "https" : "http") + "://" + this.hostname + ":" + this.extPort;
+    }
+    return undefined;
+  }
+
+  readEdgeListFromFile(): OriginsAndEdgesPerTerritory {
```
Comment (birme) [inline packages/server/src/whip/whipEndpoint.ts:156]: I suggest to return a promise (async) here as well so it aligns with the `readEdgeListFromService()` method

File: sfu-config.json [hunk 3]
```diff
@@ -1,9 +1,10 @@
-{
-    "origin": "http://localhost:8080/conferences/",
-    "edges": [
-        {
-            "sfu": "http://localhost:8081/conferences/",
-            "egress": "http://localhost:8001/api"
-        }
-    ]
+{
```
Comment (birme) [inline sfu-config.json:1]: I don't like that we are changing the format for this JSON file, why is that needed?

File: packages/server/src/server.ts [hunk 4]
```diff
@@ -1,65 +1,89 @@
...
+  https: process.env.WHIP_ENDPOINT_USE_HTTPS && process.env.WHIP_ENDPOINT_USE_HTTPS === "true",
+  tls: tlsOptions,
+  iceServers: iceServers,
+  enabledWrtcPlugins: [ "rtsp", "rtmp", "sfu-broadcaster" ], 
+  resourceManagerUrl,
+  territoryCode
+});
+
+let sfuConfigData: SfuConfigData = {};
+if(endpoint.useResourceManager()) {
```
Comment (birme) [inline packages/server/src/server.ts:54]: I don't think we need this function() in the WHIP endpoint library. You could here just check whether the `resourceManagerUrl` is not null.

File: sfu-config.json (hunk 3 above)
Comment (birme) [reply sfu-config.json:1]: On the other hand this is isolated to the reference implementation right, and then it is ok assuming no API changes has been made in `registerBroadcasterClient()`

File: packages/server/README.md [hunk 5]
```diff
@@ -1,116 +1,120 @@
...
+{
+  port: number, // port to bind to
+  extPort: number, // port that is exposed in public (default same as port)
+  interfaceIp: string, // ip to bind and listen on
+  hostname: string, // hostname or public IP
+  https: boolean, // use https
+  tls: { key: string, cert: string }, // key and cert for TLS termination (optional),
+  iceServers: [ { urls: string, username?: string, credential?: string }], // list of STUN/TURN servers
+  enabledWrtcPlugins: string[], // list of plugins to enabled. Available are "sfu-broadcaster", "rtsp", "rtmp"
+}
```
Comment (birme) [inline packages/server/README.md:80]: We should add the additional options that were added here

File: packages/server/README.md [hunk 6]
```diff
@@ -1,116 +1,120 @@
...
+    key: fs.readFileSync("./server-key.pem"),
+    cert: fs.readFileSync("./server-cert.pem"),
+  },
+  iceServers: [{ urls: "stun:stun.l.google.com:19302" }],
+  enabledWrtcPlugins: [ "sfu-broadcaster" ], 
+});
+
+// ...
+```
+
```
Comment (birme) [inline packages/server/README.md:65]: Add another example (with code) how to use the library with a resource manager.

**Review by birme — CHANGES_REQUESTED** [review]

File: sfu-config.json (hunk 3 above)
Comment (zapfire88) [reply sfu-config.json:1]: I updated the json to be more in line with the data recieved from a Resource Manager. The broadcasterClient has not been modified.

**Review by birme — CHANGES_REQUESTED** [review]
Looks good, just some small tweaks left.
I don't understand though why the diff is so big in many files?

Fix the minor comments, clean up the PR commits and then I will merge/rebase into main and close the PR

File: packages/server/README.md [hunk 7]
```diff
@@ -1,116 +1,140 @@
...
+### Environment variables
+
+The following environment variables are read to override default values:
+- `ICE_GATHERING_TIMEOUT`: (default: 4000 ms): Timeout for gathering all ICE candidates
+- `API_KEY`: Authorization key that clients must use to get ICE server config on OPTIONS request
+- `SFU_CONFIG_FILE`: Used to configure the path to the local json file. Default value is `../../sfu-config.json`.
+
+### Example environment variables in server.ts
+
+The following environment variables are used in `server.ts` as an example. They are set to enable the use of a resource manager:
```
[the remark this answers is not in the document]
Comment (zapfire88) [reply packages/server/README.md]: That's actually true! I'll remove the "header"
